src/base/LruMap.h

```diff
@@ -43,7 +56,7 @@ template <class EntryValue, size_t EntryCost = sizeof(EntryValue)> class LruMap
     /// Search for an entry, and return a pointer
     EntryValue *get(const char *key);
     /// Add an entry to the map
-    bool add(const char *key, EntryValue *t);
+    bool add(const char *key, EntryValue *t, int ttl);
```

Comment:
Two independent comments here, one responding to your comment above and the other documenting a new (to me) angle to the problem.

1. A caller that _does not have a notion of cached value TTL_ can supply nothing but a fake TTL, by definition. Yes, a caller may develop the notion of value expiration in the future, but there is no pressure to add more fake ones now, especially since changes introducing real TTL values are not imminent.
2. I just realized that LruMap already requires a default TTL parameter in the LruMap constructor. Please use that instead of the defaultTtl().

In summary, I think we should do something like this:

```
LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
{
    return addWithTtl(key, t, retention);
}
```